### PR TITLE
Adding Compaction support

### PR DESCRIPTION
**Pull request by aayushKumarJarvis**
Adding Compaction feature with tests

cc : @ashwanthkumar  please review

### CODE REVIEW

File: ops/compact.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	db, err := gorocksdb.OpenDb(opts, source)
+
+	keys.Start = db.NewIterator(compactOpts).Key().Data()
+	db.CompactRange(keys)
+
+	if err != nil {
+		return err
+	}
+
+	db.Close()
```
Comment (ashwanthkumar) [inline ops/compact.go]: @aayushKumarJarvis We need to make this defered and should be moved up closer to where we open the DB.

**Comment by aayushKumarJarvis** [comment]
@ashwanthkumar can you check now ?. Does it seem legit ?

File: ops/compact.go (hunk 1 above)
Comment (aayushKumarJarvis) [inline ops/compact.go]: @ashwanthkumar can you check now ? Does it seem legit ?

**Comment by ashwanthkumar** [comment]
LGTM - I'm waiting for the build + test to complete, once it's green. We can merge it in .

**Pull request merged**